### ISSUE TITLE
Incorrect error reported for ambiguous default hint

### ISSUE THREAD

**Issue by ohad** [edwinb/Idris2-boot#163]
Related to #162 .

# Steps to Reproduce
```
data Gnu : Type where
  G : String -> Gnu
  N : String -> Gnu
  U : String -> Gnu

||| Makes interaction a bit nicer
Show Gnu where
  show (G _) = "G"
  show (N _) = "N"
  show (U _) = "U"
  
%defaulthint
FooG : Gnu
FooG = G "g"

%defaulthint
FooN : Gnu
FooN = N "n"

FooU : Gnu
FooU = U "u"

||| We'll use this function to see what proof Idris finds.
Guess : {auto gnu : Gnu} -> String
Guess {gnu} = show gnu
```
# Expected Behavior
```
Main> Guess
(interactive):1:1--1:6:Multiple solutions found in search. Possible correct results:
	FooG
	FooN
```
# Observed Behavior
```
Main> Guess
(interactive):1:1--1:6:Can't find an implementation for Gnu
```

So while Idris reports an error as expect, it is not the expected error.

**Comment by ohad** [comment]
Updated example, as AutoSearch is now able to guess `True` as an argument to `G`.
I'm guessing it's expected not to report ambiguity on constructors, as detecting ambiguity on constructors would involve searching through all of them --- which would loop is many cases. 
Looking at the definition of `TCon` in `Core/Context.idr`, there's a (planned?) mechanism for requiring type constructors to be unique in proof search.
This means that the previous example exhibited the (incorrect) behaviour due to another bug, hopefully fixed now.

**Comment by ohad** [comment]
Fixed in 036bb851345e4e719b5d750666e73abed0a933a6 on my branch.
I want to try and fix #162 before creating a pull request.

**Comment by edwinb** [comment]
The default behaviour of `auto` is now to take the first solution, unless `[uniqueSearch]` is specified.
`%defaultHint` messes with this in any case, since they are meant as "last resort if all else fails" and really just a hack for default `Integer`. So there may well be something that needs fixing here, but I don't think it's what's reported in this issue any more! So I'll close it.
